### PR TITLE
Issue 25536 enable -disable-build-servers for pack, clean, test, msbuild

### LINKED ISSUE #25536: dotnet build -disable-build-servers is not working
My apologies; the templates are not working right now for reasons I didn't spend too long looking for. There are no errors on the javascript console.

This issue was discovered trying to track down a failure in release builds that seems to be somehow related to the build server. As often as not, running dotnet build-server shutdown from another console would clear the block. Found issue #22472 as an already completed issue so was expecting to be able to just apply the fix.

We tried `dotnet test -disable-build-servers` and it complains that it doesn't recognize it. Backtracked to `dotnet build -disable-build-servers` and discovered that doesn't work either. The error is "MSBUILD : error MSB1001: Unknown switch."

````
C>dotnet --info
.NET SDK (reflecting any global.json):
 Version:   6.0.300
 Commit:    8473146e7d

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22000
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.300\

Host (useful for support):
  Version: 6.0.5
  Commit:  70ae3df4a6

.NET SDKs installed:
  3.1.201 [C:\Program Files\dotnet\sdk]
  3.1.419 [C:\Program Files\dotnet\sdk]
  5.0.408 [C:\Program Files\dotnet\sdk]
  6.0.202 [C:\Program Files\dotnet\sdk]
  6.0.300 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.24 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.25 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.24 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.25 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.24 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.25 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download

C>
````

### PR DESCRIPTION
**Pull request by JL03-Yue**
Regarding the [issue](https://github.com/dotnet/sdk/issues/25536) 

Fixes #25536: enable -disable-build-servers for pack, clean, test, and msbuild. Added -disable-build-servers feature to pack, clean, test, and msbuild. Added options in --help for pack, clean, and test; 

Not changed the --help for msbuild

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/sdk?pullRequest=30481) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/sdk?pullRequest=30481) before we can accept your contribution.<br/><br/>:x: JL03-Yue [sign now](https://cla.dotnetfoundation.org/dotnet/sdk?pullRequest=30481)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/sdk?pullRequest=30481) it.</sub>

**Comment by dsplaisted** [comment]
@baronfel @rainersigwald @dotnet/dotnet-cli In this PR, we're adding the `--disable-build-servers` option to the `dotnet msbuild` command (as well as several other commands).  The `msbuild` command currently doesn't take any CLI-specific options, and forwards the handling of the help on to MSBuild.

So how should we handle the help text for `--disable-build-servers` with the MSBuild command.  Should we try to append it to the help message that MSBuild prints?  Right now in this PR it isn't included in the `dotnet msbuild --help` output.

**Comment by rainersigwald** [comment]
I would lean to prepend it myself, but prepend or append would be ok with me.

**Comment by marcpopMSFT** [comment]
The msbuild text is quite long. Are there any other commands we plan on adding to the msbuild Verb that we'd want in the help?  If so, it might make sense to append and have an "Additional Options" section (strange to call it additional options at the beginning is all)

File: src/Cli/dotnet/Parser.cs [hunk 1]
```diff
@@ -281,6 +281,28 @@ private static void SetHelpCustomizations(HelpBuilder builder)
                 }
             }
 
+            public void additionalOption(HelpContext context)
```
Comment (dsplaisted) [inline src/Cli/dotnet/Parser.cs:284]: Let's rename this so it's more clear what this means.  Also, method and class names should usually be "PascalCased", ie the first letter of each word capitalized.


```suggestion
            public void WriteAdditionalMSBuildOptions(HelpContext context)
```

File: src/Cli/dotnet/commands/dotnet-msbuild/Program.cs [hunk 2]
```diff
@@ -1,17 +1,47 @@
...
+        }
+
+        public static MSBuildCommand FromArgs(string[] args, string msbuildPath = null)
+        { 
+            var parser = Cli.Parser.Instance;
+            var result = parser.ParseFrom("dotnet msbuild", args);
+            return FromParseResult(result, msbuildPath);
+        }
+
+        public static MSBuildCommand FromParseResult(ParseResult parseResult, string msbuildPath = null)
```
Comment (dsplaisted) [inline src/Cli/dotnet/commands/dotnet-msbuild/Program.cs:26]: @baronfel Just wondering, did you review the changes to this class?  I think these changes make it more like the other command classes we have, so it should be fine, but wouldn't hurt to have you take a look.

File: src/Tests/dotnet.Tests/dotnet-msbuild/GivenDotnetMSBuildInvocation.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    public class GivenDotnetMSBuildInvocation : IClassFixture<NullCurrentSessionIdFixture>
+    {
+        const string ExpectedPrefix = "-maxcpucount -verbosity:m";
+       
+        private static readonly string WorkingDirectory =
+            TestPathUtilities.FormatAbsolutePath(nameof(GivenDotnetPackInvocation));
+
+        [Theory]
+        [InlineData(new string[] { "--disable-build-servers" }, "-p:UseRazorBuildServer=false -p:UseSharedCompilation=false /nodeReuse:false")]
+
```
Comment (dsplaisted) [inline src/Tests/dotnet.Tests/dotnet-msbuild/GivenDotnetMSBuildInvocation.cs:23]: Delete this blank line between the method and its attributes (Theory and InlineData).

**Review by dsplaisted — APPROVED** [review]

File: src/Cli/dotnet/commands/dotnet-msbuild/Program.cs (hunk 2 above)
Comment (baronfel) [reply src/Cli/dotnet/commands/dotnet-msbuild/Program.cs:26]: Not too deeply, no - I've mainly looked at the actual code for emitting the new options section. I can take another pass at the broader set of changes though!

**Review by baronfel — APPROVED** [review]

**Pull request merged**